### ISSUE TITLE
Add is unnecessary in external Commit

### ISSUE THREAD

**Issue by bifurcation** [mlswg/mls-protocol#505]
When a new joiner sends an external Commit, within the current structure, they have to generate and send **two** KeyPackages:  First in an Add proposal, and second in `Commit.path.leaf_key_package`.  Obviously, this is wasteful, since the second immediately overwrites the first.  It also complicates receiver processing, since the receiver has to find where the appropriate Add happened so that it knows the sender's index in the new tree, so that it knows how to apply the update path in the Commit.

We should remove the requirement for an Add commit in an external Commit, and instead assign the new member the leftmost free slot after all proposals have been applied, then apply the commit from there.

**Comment by raphaelrobert** [comment]
I agree that the key package in the Add proposal is redundant.
I'm not so sure it will simplify implementations so much, because key packages from the update path of a Commit are usually treated like Update proposals (e.g. the constraint that the identity must be the same between the old and new credential).
Tl;DR: I agree with the change, but I want to raise awareness about issues this introduces.
